tests/app/views/test_suppliers.py

```diff
@@ -1427,89 +1520,225 @@ def test_countersigned_at_timestamp_cannot_be_set(self):
 
-        assert_equal(response.status_code, 200)
+        assert response.status_code == 200
         data = json.loads(response.get_data())
-        assert_equal(data['frameworkInterest']['agreementDetails'], agreement_details_payload)
+        assert data['frameworkInterest']['agreementDetails'] == agreement_details_payload
 
         # while we're at it let's test the agreementDetails partial updating behaviour
         agreement_details_update_payload = {
-            "other": {
-                "json": 456,
-            },
-            "here": None,
+            "uploaderUserId": 1,
         }
         response2 = self.supplier_framework_update(
-            0, 'digital-outcomes-and-specialists',
-            update={'agreementDetails': agreement_details_update_payload})
+            0, 'g-cloud-8',
+            update={
+                'agreementReturned': True,
+                'agreementDetails': agreement_details_update_payload
+            }
+        )
 
         agreement_details_payload.update(agreement_details_update_payload)
-        # json validator should strip this key
-        del agreement_details_payload["here"]
+        assert response2.status_code == 200
+        data2 = json.loads(response2.get_data())
+        assert data2['frameworkInterest']['agreementDetails'] == {
+            "signerName": "name",
+            "signerRole": "role",
+            "uploaderUserId": 1,
+            "uploaderUserName": "my name",
+            "uploaderUserEmail": "test+1@digital.gov.uk",
+            "frameworkAgreementVersion": "v1.0",
+        }
+
+    def test_can_not_set_agreement_details_on_frameworks_without_framework_agreement_version(self):
+        agreement_details_payload = {
+            "signerName": "name",
+            "signerRole": "role",
+            "uploaderUserId": 1,
+        }
+        response = self.supplier_framework_update(
+            0, 'digital-outcomes-and-specialists',
+            update={'agreementDetails': agreement_details_payload})
+
+        # weird because it still returns a 200-level response even though it doesn't update anything
+        assert response.status_code == 200
+        data = json.loads(response.get_data())
+        assert data['frameworkInterest']['agreementDetails'] is None
+
+    def test_setting_agreement_details_with_nonexistent_user_id_doesnt_return_user_details(self):
+        agreement_details_payload = {
+            "signerName": "name",
+            "signerRole": "role",
+            "uploaderUserId": 999
+        }
+        response = self.supplier_framework_update(
+            0, 'g-cloud-8',
+            update={'agreementDetails': agreement_details_payload})
+
+        data = json.loads(response.get_data())
+        assert response.status_code == 200
+        assert data['frameworkInterest']['agreementDetails'] == {
+            "signerName": "name",
+            "signerRole": "role",
+            "uploaderUserId": 999
```

Comment:
Yeah this seems weird - if the user ID doesn't exist in our system we should complain I think.